solr/solrj/src/resources/java-template/api.mustache

```diff
@@ -205,26 +203,23 @@ public class {{classname}} {
 
             @Override
             @SuppressWarnings("unchecked")
-            public RequestWriter.ContentWriter getContentWriter(String expectedType) {
+            public RequestWriter.ContentWriter getContentWriter(String _expectedTypeIGNORE) {
```

Comment:
`getContentWriter` is passed an "expectedType" from the `SolrClient.requestWriter` which defaults to "javabin".  But V2 should ignore that requestWriter as it's opinionated on how to write -- only supports JSON via Jackson or raw writing.  Perhaps that field in SolrClient should be named `defaultRequestWriter`... although we're not choosing a different RequestWriter so.... :shrug:

Reply:
[0] Maybe some version of this rationale should live as a comment in the code/template, so that we can notice when those preconditions/assumptions change?

e.g.

> // v2 currently only supports reading JSON request-bodies, so we ignore the expected type suggestion provided by callers.